notion-server/src/ethereum/mod.rs

```diff
@@ -403,21 +436,45 @@ impl ShardChain {
     }
 }
 
+trait ToBytes32 {
+    fn to_bytes32(&self) -> Result<[u8; 32]>;
+}
+
+impl ToBytes32 for Vec<u8> {
+    fn to_bytes32(&self) -> Result<[u8; 32]> {
+        if self.len() == 32 {
+            let mut ret: [u8; 32] = [0; 32];
+            ret.copy_from_slice(&self[..]);
+            Ok(ret)
+        } else {
+            Err(Error::InvalidBytes32)
+        }
+    }
+}
+
 #[derive(Debug, Default, Hash, Clone, Copy, Eq, PartialEq)]
 pub struct EeIndex(u32);
 
 // The execution environment data that lives on the beacon chain
 // Does NOT include shard-specific EE state
 #[derive(Debug)]
 struct ExecutionEnvironment {
-    wasm_code: Vec<u8>,
+    pub wasm_code: Vec<u8>,
```

Comment:
just curious -- is it necessary to make these `pub`?  I could be wrong but I thought that sub-modules have access to private members of their parent modules, so I think this should still work with `private` fields.  If not, could you point out which lines require this `pub` switch?

Reply:
You are indeed correct, I'll revert that change.